### DISCUSSION TITLE
Environment variables not inherited by Gradle child process

### DISCUSSION THREAD

**Discussion by christopherfrieler** [renovatebot/renovate#28437] [Request Help]
### What would you like help with?

I think I found a bug

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitLab, also self-hosted

### Please tell us more about your question or problem

It looks like I came across the bug from #5456.

I have a gradle project that requires some Artifactory credentials as project properties. I set these as environment variables with prefix "ORG_GRADLE_PROJECT_" in the 'customEnvVariables' of my renovate config. They also show up in the logged command config (see logs). However, the gradle build complains about the missing project properties.

The renovate version should be the latest, as we use the "renovate/renovate:slim" docker image.

One difference to #5456 is that I use 'customEnvVariables' as 'trustLevel' is deprecated. But as far as I understand, this is exactly how i should work. I also tried to set 'exposeAllEnv', which added a lot more variables from the CI environment, but gradle didn't pick them up. Can it be that this broke when changing 'trustLevel' to 'customEnvVariables'?

### Logs (if relevant)

<details><summary>Logs</summary>

```

WARN: Error executing gradle wrapper update command. It can be not a critical one though. (repository=XXXXX, branch=renovate/gradle-8.x)
       "err": {
         "cmd": "/bin/sh -c ./gradlew :wrapper --gradle-distribution-url https://services.gradle.org/distributions/gradle-8.7-bin.zip",
         "stderr": "\nFAILURE: Build completed with 2 failures.\n\n1: Task failed with an exception.\n-----------\n* Where:\nBuild file 'XXXXX/build.gradle.kts' line: 31\n\n* What went wrong:\nCould not get unknown property 'artifactory.user' for root project 'XXXXX' of type org.gradle.api.Project.\n\n ...",
         "stdout": "XXXXX",
         "options": {
           "cwd": "XXXXX",
           "encoding": "utf-8",
           "env": {
             "GRADLE_OPTS": "-Dorg.gradle.parallel=true -Dorg.gradle.configureondemand=true -Dorg.gradle.daemon=false -Dorg.gradle.caching=false",
             "HOME": "/home/ubuntu",
             "PATH": "/home/ubuntu/.cargo/bin:/home/ubuntu/.local/bin:/go/bin:/home/ubuntu/bin:/home/ubuntu/.npm-global/bin:/home/ubuntu/.cargo/bin:/home/ubuntu/.local/bin:/go/bin:/home/ubuntu/bin:/home/ubuntu/.npm-global/bin:/home/ubuntu/.cargo/bin:/home/ubuntu/.local/bin:/go/bin:/home/ubuntu/bin:/home/ubuntu/.npm-global/bin:/home/ubuntu/.cargo/bin:/home/ubuntu/.local/bin:/go/bin:/home/ubuntu/bin:/home/ubuntu/.npm-global/bin:/home/ubuntu/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
             "LC_ALL": "C.UTF-8",
             "LANG": "C.UTF-8",
             "DOCKER_HOST": "tcp://docker:2376",
             "DOCKER_TLS_VERIFY": "1",
             "DOCKER_CERT_PATH": "/certs/client",
             "ORG_GRADLE_PROJECT_artifactory.user": "XXXXX",
             "ORG_GRADLE_PROJECT_artifactory.password": "**redacted**",
             "CONTAINERBASE_CACHE_DIR": "XXXXX/.renovate/cache/containerbase"
           },
           "timeout": 900000
         },
         "exitCode": 1,
         "name": "ExecError",
         "message": "Command failed: ..."
       }

```

</details>

**Comment by rarkins** [comment]
The logs indicate that the env is passed through:

```
             "ORG_GRADLE_PROJECT_artifactory.user": "XXXXX",
             "ORG_GRADLE_PROJECT_artifactory.password": "**redacted**",
```

You should try with exposeAllEnv=true first
